DEVELOPER_GUIDE.md

```diff
@@ -1,29 +1,44 @@
+Due to that, they don't carry any effect on implementation as they are just arbitrary labels for the generic type parameters, useful to communicate the intention to the developers and users of the library. Still F# type system will normally show some constraints at the method level, but these constraints are sometimes not very helpful in communicating the related abstraction.
 
-Those _Type Class Implementation Function Types_ define static methods of several kinds:
+The concrete implementation of these abstractions on the other hand is done, minimally, by defining a concrete type (an Invokable) per function.
+
+For example, one of the simplest abstractions `Monoid` has its core implementation split under two types `Plus` and `Zero`.
+
+Those Invokers contains static methods of several kinds:
+
+* concrete overloads, for example Monoid's `Plus` will have overloads of a `Plus` method (named the same as the type defining those) for each of the BCL and FSharp.Core types that ought to support the `Monoid` type class, e.g. one to concatenate two strings, one to concatenate two lists, one to concatenate two arrays, and so on and so forth.
+* an _invokable_ method, that should always be defined as `inline`, which embeds the SRTP dispatcher machinery that is used to pick the right overload (whether defined in F#+ or outside of it)
+* a default or set of default implementations, those may need to be defined in an instrinsic type extension due to F# compiler implementation details to enable being picked up correctly for types that end up supporting the abstraction when such abstraction comes with a default implementation expressed in terms of composing several functions, those are always defined as `inline`.
 
-* concrete overloads, for exemple Monoid's `Plus` will have overloads of a `Plus` method (named the same as the type defining those) for each of the BCL and FSharp.Core types that ought to support the `Monoid` type class, e.g. one to concatenate two strings, one to concatenate two lists, one to concatenate two arrays, and so on and so forth.
-* an _invoker_ method, that should always be defined as `inline`, which embeds the SRTP machinery that is used to pick the right overload (whether defined in F#+ or outside of it for implementing type class)
-* a default or set of default implementations, those may need to be defined in a type extension due to F# compiler implementation details to enable being picked up correctly for types that end up supporting the type class when such type class come with default implementation expressed in terms of composing several functions, those are also always defined as `inline`.
 
-### Implementation idioms
 
 #### Concrete implementations
 
+We should add as much concrete implementations for primitive types as possible, since the end user won't be able to add them later.
```

Comment:
I wonder what `primitive types` means in this context. 
Googling "F# primitive types" returns docs: https://docs.microsoft.com/pl-pl/dotnet/fsharp/language-reference/basic-types
I guess that what you've meant here was FSharp.Core types, but I may be wrong.

Reply:
In fact I mean both .NET framework types and F# core types, which are the sole dependencies of this library. Not sure what's the best word for describing them.

Reply:
I think you've already referenced them in this guide as `BCL and FSharp.Core types`, which I find accurate.